### PR TITLE
Detecting root from runnable path to make the ext vscode workspace agnostic

### PR DESCRIPTION
**Pull request by c0nfleis**
This PR allows an end user to not have to open VSCode in a specific working directory in order to use their runnables with shared common logic.

### CODE REVIEW

File: src/web/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    Examples:
+    "/windmill/under/windmill-src/f/testing/bar.py" => "/windmill/under/windmill-src"
+    "/windmill/free/windmill-src/u/ryan/bar.py" => "/windmill/free/windmill-src"
+*/
+export function getRootPathFromRunnablePath(fullPath: string): string {
+    const dirs = ["/u/", "/f/"];
+
+    for (const dir of dirs) {
+        if (fullPath.includes(dir)) {
+            fullPath = fullPath.endsWith(dir) ? fullPath.split(dir)[0] : fullPath.split(`${dir}`)[0];
```
Comment (rubenfiszel) [inline src/web/helpers.ts:14]: I think the 2 branches are doing the same thing :)

**Comment by rubenfiszel** [comment]
This is great, thanks a lot. 

You mentioned adding jest, that would be great! But we can do it in a separate PR

**Comment by rubenfiszel** [comment]
Btw have you been able to test/run the vscode extension ?

**Comment by c0nfleis** [comment]
> Btw have you been able to test/run the vscode extension ?

Yes! been running the extension locally using the readme. Just pressing F5 a whole lot haha!

File: src/web/extension.ts [hunk 2]
```diff
@@ -337,7 +339,7 @@ export function activate(context: vscode.ExtensionContext) {
           case "flow":
             // channel.appendLine("flow message");
             let uri = vscode.Uri.parse(message.uriPath);
-            if (!message.uriPath.endsWith("flow.yaml")) {
+            if (!message.uriPath?.endsWith("flow.yaml")) {
```
Comment (c0nfleis) [inline src/web/extension.ts:342]: There was an error happening in main that I found and squashed using this! 


This was the error:
```
Uncaught TypeError TypeError: Cannot read properties of undefined (reading 'endsWith')
    at eval (/Users/bmaldonado/Code/NonPanther/windmill-vscode/src/web/extension.ts:340:34)
```

**Comment by rubenfiszel** [comment]
Thanks a lot, will do the necessary work to publish it

**Pull request merged**